nixos/modules/services/hardware/nvidia-container-toolkit/default.nix

```diff
@@ -129,137 +128,191 @@
           '';
         };
       };
-
     };
 
-  config = lib.mkIf config.hardware.nvidia-container-toolkit.enable {
-    assertions = [
-      {
-        assertion =
-          config.hardware.nvidia.datacenter.enable
-          || lib.elem "nvidia" config.services.xserver.videoDrivers
-          || config.hardware.nvidia-container-toolkit.suppressNvidiaDriverAssertion;
-        message = ''`nvidia-container-toolkit` requires nvidia drivers: set `hardware.nvidia.datacenter.enable`, add "nvidia" to `services.xserver.videoDrivers`, or set `hardware.nvidia-container-toolkit.suppressNvidiaDriverAssertion` if the driver is provided by another NixOS module (e.g. from NixOS-WSL)'';
-      }
-      {
-        assertion =
-          ((builtins.length config.hardware.nvidia-container-toolkit.csv-files) > 0)
-          -> config.hardware.nvidia-container-toolkit.discovery-mode == "csv";
-        message = ''When CSV files are provided, `config.hardware.nvidia-container-toolkit.discovery-mode` has to be set to `csv`.'';
-      }
-    ];
+  config = lib.mkMerge [
+    (lib.mkIf config.virtualisation.docker.enableNvidia {
+      environment.etc."nvidia-container-runtime/config.toml".text = ''
+        disable-require = true
+        supported-driver-capabilities = "compat32,compute,display,graphics,ngx,utility,video"
+        [nvidia-container-cli]
+        environment = []
+        ldconfig = "@${lib.getExe' pkgs.glibc "ldconfig"}"
+        load-kmods = true
+        no-cgroups = false
+        path = "${lib.getExe' pkgs.libnvidia-container "nvidia-container-cli"}"
+        [nvidia-container-runtime]
+        mode = "auto"
+        runtimes = ["docker-runc", "runc", "crun"]
+        [nvidia-container-runtime-hook]
+        path = "${lib.getOutput "tools" pkgs.nvidia-container-toolkit}/bin/nvidia-container-runtime-hook"
+        skip-mode-detection = false
+        [nvidia-ctk]
+        path = "${lib.getExe' pkgs.nvidia-container-toolkit "nvidia-ctk"}"
+      '';
+
+      virtualisation.docker = {
+        daemon.settings = {
+          default-runtime = "nvidia";
+          runtimes.nvidia = {
+            path = "${lib.getOutput "tools" config.hardware.nvidia-container-toolkit.package}/bin/nvidia-container-runtime";
```

Comment:
Why use `config.hardware.nvidia-container-toolkit.package` here but `pkgs.nvidia-container-toolkit` above? Wouldn't we want to use `config.hardware.nvidia-container-toolkit.package` throughout?